js/adapt-contrib-spoor.js

```diff
@@ -78,6 +78,14 @@ define([
                 if(settings.hasOwnProperty("_commitRetryDelay")) {
                     scorm.commitRetryDelay = settings._commitRetryDelay;
                 }
+
+                if (settings.hasOwnProperty("_exitStateIfIncomplete")) {
```

Comment:
this is probably a good opportunity to stop abusing `hasOwnProperty` in this way! 

@oliverfoster sorry I know you've covered this before but what's the best way to check for a property when it could have a legitimately 'falsy' value like `""` or `0`? Would something like `if ('_exitStateIfIncomplete' in settings)` be more appropriate?

Reply:
```js
// when property value is checked and is not a falsy
if (object.attributeName) { 
 // value can be anything but: not "", 0, null, undefined or NaN
} 

// when instance property should exist and not be inherited from the prototype chain
if (object.hasOwnPropert(attributeName)) {
  // property must have been set on the instance
  // value can be anything
}

// when the property should exist and can be inherited
if (attributeName in object) {
  // property can have been inherited from the prototype chain
  // value can be anything
}
```
### References

* [in operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in)
* [hasOwnProperty function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty)
* [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)
* [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy)

Reply:
Thanks @oliverfoster, very helpful!

@canstudios-matth the checks for `_suppressErrors` and `_commitOnStatusChange` can be changed to `if (settings._suppressErrors)` and `if (settings._commitOnStatusChange)`. 

As properties `_timedCommitFrequency`, `_maxCommitRetries` & `_commitRetryDelay` should all be numerical (with `0` as a valid value), some sort of check for a numerical value would be good, perhaps `_.isFinite`?

As the new settings can have `""` as a valid value they can either be left as `hasOwnProperty` or changed to use the `in` operator, as you prefer.

Reply:
```js
if (object.property || object.property === 0) {

}

if (_.isNumber(object.property)) {

}
```

Either of the above is good. JSON can't represent non-finite numbers. Positive and negative floating point integers and scientific notation only I believe. Such that:
```js
JSON.parse("Infinity"); // throws an error
JSON.parse("3.141e7"); // returns 31410000
JSON.stringify(Infinity); // returns null
```

#### References:
https://underscorejs.org/#isNumber
https://underscorejs.org/#isFinite
https://www.json.org/

Reply:
@oliverfoster my main reason for suggesting `_.isFinite` rather than `_.isNumber` is that the latter will return `true` for `NaN` whereas the former will not. But perhaps it doesn't really matter.